multiversx_sdk/core/transactions_outcome_parsers/resources.py

```diff
@@ -60,7 +60,19 @@ def find_events_by_identifier(transaction_outcome: TransactionOutcome, identifie
 
 
 def find_events_by_first_topic(transaction_outcome: TransactionOutcome, topic: str) -> List[TransactionEvent]:
-    return find_events_by_predicate(transaction_outcome, lambda event: event.topics[0].decode() == topic if len(event.topics) else False)
+    def is_topic_matching(event: TransactionEvent):
+        if not len(event.topics):
+            return False
+
+        try:
+            decoded_topic = event.topics[0].decode()
+            if decoded_topic == topic:
+                return True
+            return False
+        except:
```

Comment:
Optionally, can nominate the specific exception e.g. `UnicodeDecodeError`.

Reply:
done